torchtune/utils/_compile_utils.py

```diff
@@ -0,0 +1,24 @@
+import torch
+
+def wrap_compile(model):
+    """
+    Wraps the model with torch.compile. This function will also
+	register a state_dict post hook that allows state_dicts produced
+	with torch.compile training to behave as regular eager mode models.
+	In particular, it strips away a torch.compile specific prefix
+	added to the state_dict by torch.compile.
+
+	Args:
+		model (nn.Module): model to wrap with compile.
+    """
+    model = torch.compile(model)
+    model._register_state_dict_hook(_consume_torch_compile_prefix)
+    return model
+
+
+def _consume_torch_compile_prefix(model, state_dict, *args, prefix="_orig_mod.", **kwargs):
```

Comment:
don't need args or kwargs?

Reply:
Kind of unfortunate but don't know how to work around this. The reason I have it is because this runs as a state_dict hook, so nn.Module implementation invokes these with additional arguments (such as the model, maybe additional metadata) that we don't need or use. So I just hide these in args, kwargs. Maybe could call them (unused_args, unused_kwargs) to make it more explicit?

Reply:
A couple questions here: 

(1) does prefix ever vary? If this is specific to compile do we actually need to parametrize it? (And if we do, are we able to move it out from in between args and kwargs)
(2) Anything else we have to watch out for here in terms of composability with other functionality (specifically other state dict hooks I guess)? I know you mentioned there are still some gaps with distributed, is this one of the rough edges there?

Reply:
`prefix` should remain constant modulo BC breaking changes from pytorch.

Re: composability with other state_dict hooks, the main thing to watch out for is hook execution order. If there is a wrapper such as DDP, that also adds prefix to modules, and this hook runs before DDP's state_dict hook, it won't remove any of the prefixes since the name begins with DDP's prefix and not torch compile prefix.

If we hit this issue though, checkpointing issues will be raised, so we'd be aware that way.